### DISCUSSION TITLE
Prevent the scroll bar from auto widening on mouse hover

### DISCUSSION THREAD

**Discussion by josephnarai** [AvaloniaUI/Avalonia#6718] [Q&A]
Is there a way to keep the scroll bar narrow so it does not auto expand when you hover the mouse over it?

**Comment by maxkatz6** [comment]
This property is what you need 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ScrollViewer.cs#L415

**Reply by maxkatz6** [reply]
Oh, you actually need the opposite. It's a bit complicated.

ScrollBar has IsExpanded property, but it's readonly and you can't change it.
Instead, you can create your own styles to make it look like always collapsed. For reference see https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ScrollBar.xaml

Remember, if you have referenced Fluent theme, all default styles will be applied and your styles will be applied after overriding previous. So in this case default logic for collapsing will be still there.
You can start by creating these styles:
```
  <Style Selector="ScrollBar:vertical /template/ Thumb.thumb">
    <Setter Property="RenderTransform" Value="{DynamicResource VerticalSmallScrollThumbScaleTransform}" />
  </Style>
  <Style Selector="ScrollBar:horizontal /template/ Thumb.thumb">
    <Setter Property="RenderTransform" Value="{DynamicResource HorizontalSmallScrollThumbScaleTransform}" />
  </Style>
```

**Reply by josephnarai** [reply]
<TransformOperations x:Key="VerticalSmallScrollThumbScaleTransform">scaleX(0.5) translateX(-2px)</TransformOperations>
    <TransformOperations x:Key="HorizontalSmallScrollThumbScaleTransform">scaleY(0.5) translateY(-2px)</TransformOperations>
Thanks, As a quick fix, I ended up just overriding:

```
    <x:Double x:Key="ScrollBarSize">6</x:Double>
    <x:Double x:Key="ScrollBarButtonArrowIconFontSize">4</x:Double>
    <TransformOperations x:Key="VerticalSmallScrollThumbScaleTransform">scaleX(0.5) translateX(-2px)</TransformOperations>
    <TransformOperations x:Key="HorizontalSmallScrollThumbScaleTransform">scaleY(0.5) translateY(-2px)</TransformOperations>

```
    This still has the transition to the square scrollbar on hover, but it's only a little bigger and still fits in...

    I'll see if I can keep the style of the collapsed version by overrideing some parametersin ScrollBar

**Reply by josephnarai** [reply]
How do I mark your reply to your first post as the answer?
